### ISSUE TITLE
webm files do not play properly.

### ISSUE THREAD

**Issue by CornFromHell** [mpv-player/mpv#13838]
![SpeechlessCat](https://github.com/mpv-player/mpv/assets/147477877/a362a3c8-401a-4d14-ab0a-431e4f35e9c9) 
![2024-04-08_222756](https://github.com/mpv-player/mpv/assets/147477877/a7a353f5-ac68-417c-af06-93e803075f08)
[SpeechlessCat.zip](https://github.com/mpv-player/mpv/files/14906076/SpeechlessCat.zip)

ffmpeg -i ./SpeechlessCat.gif -c:v libvpx-vp9 -crf 12 -b:v 500K ./SpeechlessCat.webm

When I encode a gif file into webm(VP9) using ffmpeg, MPV cannot play the webm file properly. It just outputs a blue screen. The webm file plays normally in VLC, Firefox, or Chrome.

Information:

Windows Version: Windows 10 Pro 64bit intel (v10 b19045)
GPU model, driver and version: NVIDIA GT 1030, v31.0.15.3623
mpv version: 0.36.0-274 (Sep 2023)
Source of the mpv binary: https://sourceforge.net/projects/mpv-player-windows/files/64bit-v3/
ffmpeg version: 2023-07-19-git-efa6cec759-full_build-www.gyan.dev Copyright (c) 2000-2023 the FFmpeg developers built with gcc 12.2.0 (Rev10, Built by MSYS2 project)

**Comment by CounterPillow** [comment]
where log file

**Comment by CornFromHell** [comment]
@zhengqwe

hwdec=auto <-I checked now and deleting the hardware acceleration settings solved the problem. One thing that puzzles me is that this problem only occurs when the GIF is encoded with VP9. I don't know why this error occurs.

**Comment by CornFromHell** [comment]
@CounterPillow Here's that log file.
[MPV_log.txt](https://github.com/mpv-player/mpv/files/14908524/MPV_log.txt)

**Comment by CounterPillow** [comment]
```
[   0.711][f][vo/gpu/d3d11va] Could not create shader resource texture
[   0.711][e][vo/gpu] Initializing texture for hardware decoding failed.
```

Two possibilities here:
1. driver advertises support for a texture format it then doesn't actually implement and simply errors on.
2. A bug in mpv's vo_gpu hwdec interop. Try with an up-to-date version of mpv and `--vo=gpu-next`.

**Comment by CornFromHell** [comment]
@CounterPillow
![2024-04-09_020037](https://github.com/mpv-player/mpv/assets/147477877/8969f24c-bb29-4bba-b7ae-67d87f1a66e5)

I tried "--vo=gpu-next" and MPV output a purple screen. -_-;

**Comment by kasper93** [comment]
Could you try if it decodes in ffmpeg directly? Check the out.mp4 and the log if it actually did hwdec.
```
ffmpeg -v trace -hide_banner -hwaccel d3d11va -i SpeechlessCat.webm out.mp4
```
It looks like for whatever reason decoder fails to allocate surface for output.

**Comment by CornFromHell** [comment]
@kasper93 There is no problem when playing other webm (VP9) files. The problem occurs only when converting "GIF" to "VP9" codec video. This problem does not exist in other video players, and the problem is solved by deleting “hwdec=auto” in “mpv.conf”.

**Comment by kasper93** [comment]
Yes, but the question was does it work using ffmpeg cli?

**Comment by CornFromHell** [comment]
@kasper93 I tried it with ffmpeg. I got an mp4 file of 0 kilobytes.

**Comment by kasper93** [comment]
You could try reporting it to https://trac.ffmpeg.org/ in this case, because it seems to be not related to mpv.

**Comment by CornFromHell** [comment]
@kasper93 I suspected FFMPEG, but there were no problems with the SpeechlessCat.webm file in other video players or web browsers. Should I update ffmpeg to the latest?

**Comment by kasper93** [comment]
Well, yes, you should test latest version of both mpv and ffmpeg.

**Comment by CornFromHell** [comment]
@kasper93 I tried updating both MPV and FFMPEG to the latest versions, but the same problem occurs. I'll just focus on other things for now and try again later. ;(

**Comment by llyyr** [comment]
> I found out that MPV cannot handle odd numbered resolutions when the GPU acceleration option is enabled.

Sounds like an Nvidia bug

**Comment by CatKasha** [comment]
Same issue on AMD gpu, RX 6600, 24.3.1, mpv-x86_64-v3-20240414-git-6a8b130
Blue on `--hwdec=auto --vo=gpu`, purple on `--hwdec=auto --vo=gpu-next`
Interestingly the video works fine with `--gpu-api=vulkan`

**Comment by llyyr** [comment]
Can't reproduce with `--no-config --hwdec=auto` on the same gpu. On mesa master though

**Comment by Tsuchimikado** [comment]
Can confirm on built-on UHD Graphics 31.0.101.2127 Windows 10 https://sourceforge.net/projects/mpv-player-windows/files/release/mpv-0.38.0-x86_64.7z/download
mpv is blue/purple when using dxva2/d3d11va, but dxva2-copy/d3d11va-copy is normal

**Comment by Tsuchimikado** [comment]
Seems to be fixed in [mpv-x86_64-20250309-git-edf4fdf](https://sourceforge.net/projects/mpv-player-windows/files/64bit/mpv-x86_64-20250309-git-edf4fdf.7z/download)
